### PR TITLE
CLIN-1994 Adjust genotype instead of filtering

### PR DESCRIPTION
**Pull request by jecos**
Before :
Genotype with low AD was filter out before inheritance and transmissions calcultaions. It cause a lot of null values for these 2 columns

After :
Instead of filtering, we adjust genotypes. If AD is too low, we replace genotype by ./. (-1,-1). Inheritance and transmissions then consider these unknown genotypes instead of null.Rows are then filter out because there is no alternative allele.

### CODE REVIEW

File: src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala [hunk 1]
```diff
@@ -78,7 +78,19 @@ class SNV(batchId: String)(implicit configuration: Configuration) extends Occurr
...
-  val includeFilter: Column = (col("has_alt") and col("ad_alt") >= 3) or (not(col("has_alt")) and col("ad_ref") >= 3)
+  /**
+   * This column is used to adjust the genotype of a variant. It considers the following rules:
+   * - If the variant is HOM or HET and the AD_ALT is less than 3, then the genotype is set to -1/-1
+   * - If the variant is HEM and the AD_ALT is less than 3, then the genotype is set to -1
+   * - If the variant is WT and the AD_REF is less than 3 and the number of alleles is 1, then the genotype is set to -1
+   * - If the variant is WT and the AD_REF is less than 3 and the number of alleles is > 1, then the genotype is set to -1/-1
+   * - Otherwise, the genotype is set to the original value
+   */
+  private val ajustedGenotype: Column = when(col("zygosity").isin("HOM", "HET") and col("ad_alt") < 3, array(lit(-1), lit(-1)))
```
Comment (ethienneroy) [inline src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala:89]: `ajustedGenotype` should be `adjustedGenotype`

File: src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala [hunk 2]
```diff
@@ -116,8 +128,10 @@ object SNV {
       .withColumn("batch_id", lit(batchId))
       .withColumn("last_update", current_date())
       .withColumn("variant_type", lit("germline"))
+      .withColumn("zygosity", zygosity(col("calls"))) // we temporary calculate zygosities for adjusting calls column
+      .withColumn("calls", ajustedGenotype)
```
Comment (ethienneroy) [inline src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala:132]: `adjustedGenotype`

File: src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala (hunk 1 above)
Comment (jecos) [reply src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala]: fixed, ty

**Review by creativeyann17 — APPROVED** [review]

File: src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala [hunk 3]
```diff
@@ -78,7 +78,19 @@ class SNV(batchId: String)(implicit configuration: Configuration) extends Occurr
...
+  /**
+   * This column is used to adjust the genotype of a variant. It considers the following rules:
+   * - If the variant is HOM or HET and the AD_ALT is less than 3, then the genotype is set to -1/-1
+   * - If the variant is HEM and the AD_ALT is less than 3, then the genotype is set to -1
+   * - If the variant is WT and the AD_REF is less than 3 and the number of alleles is 1, then the genotype is set to -1
+   * - If the variant is WT and the AD_REF is less than 3 and the number of alleles is > 1, then the genotype is set to -1/-1
+   * - Otherwise, the genotype is set to the original value
+   */
+  private val adjustedGenotype: Column = when(col("zygosity").isin("HOM", "HET") and col("ad_alt") < 3, array(lit(-1), lit(-1)))
+    .when(col("zygosity") === "HEM" and col("ad_alt") < 3, array(lit(-1)))
```
Comment (laurabegin) [inline src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala:90]: Si le variant est HEM et que AD_ALT < 3, ça va toujours être -1/-1. Le deuxième when ne sera pas évalué.

**Review by laurabegin — CHANGES_REQUESTED** [review]

File: src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala (hunk 3 above)
Comment (laurabegin) [reply src/main/scala/bio/ferlab/clin/etl/normalized/SNV.scala:90]: lol oublie ça

**Review by laurabegin — APPROVED** [review]

**Pull request merged**